parachain/pallets/bridge/src/lib.rs

```diff
@@ -75,18 +77,51 @@ decl_module! {
 			T::Verifier::verify(who, app_id, &message)?;
 			Self::dispatch(app_id.into(), &message)
 		}
+
+		/// Submit multiple messages for dispatch to multiple target applications.
+		#[weight = 0]
+		pub fn submit_bulk(origin, messages_by_app: Vec<Messages>) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			T::Verifier::verify_bulk(who, messages_by_app.as_slice())?;
+
+			let handlers: Vec<fn(&[u8]) -> DispatchResult> = messages_by_app.iter()
+				.filter_map(|messages| Self::get_handler(messages.0.into()))
+				.collect();
+			if handlers.len() < messages_by_app.len() {
+				return Err(Error::<T>::AppNotFound.into());
+			}
+
+			let errors: Vec<DispatchError> = messages_by_app.iter()
+				.enumerate()
+				.map(|(i, messages)| {
+					let handler = handlers[i];
+					messages.1.iter()
+						.map(|msg| handler(&msg.payload))
+						.collect::<Vec<DispatchResult>>()
+				})
+				.flatten()
+				.filter_map(|r| r.err())
+				.collect();
+
+			if errors.is_empty() { Ok(()) } else { Err(Error::<T>::SkippedFailedMessages.into()) }
```

Comment:
I don't think there's much value in returning a `SkippedFailedMessages` error. Its not really actionable, especially since we can't specify the specific underlying app error(s) that occurred.

Also, users who really want to know whether their message submission failed could always query storage to see whether the apps failed to emit an event.

I would consider just returning `Ok(())`, irrespective of whether we got errors from individual apps.

Reply:
> Also, users who really want to know whether their batch message submission failed could always query storage to see whether the apps emitted an event or not

Makes sense to me. I'll change